include/nbl/video/IPhysicalDevice.h

```diff
@@ -458,6 +213,39 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             return (hostWritable & hostReadable) & (~deviceLocal);
         }
 
+        /* ImageFormatProperties2
+                !! Exposing this ImageFormatProperties is not straightforward 
+                !! because it depends on multiple parameters other than `format` including
+                !! type, tiling, usage, flags and also whatever you put in it's pNextchain
+                !! basically these could be only queried if we knew all the creation params of an image
+
+                - VkAndroidHardwareBufferUsageANDROID,
+                - VkExternalImageFormatProperties,
+                - VkFilterCubicImageViewImageFormatPropertiesEXT,
+                - VkImageCompressionPropertiesEXT,
+                - VkSamplerYcbcrConversionImageFormatProperties,
+                - VkTextureLODGatherFormatPropertiesAMD
+
+            !! Same goes for `vkGetPhysicalDeviceSparseImageFormatProperties2`
+        */
+
+        /* FormatProperties2 
+                - VkDrmFormatModifierPropertiesListEXT(linux stuff)
+                - VkDrmFormatModifierPropertiesList2EXT(linux stuff)
+                - VkFormatProperties3: (available in Vulkan Core 1.1)
+                    Basically same as VkFromatProperties but the flag type is VkFormatFeatureFlagBits2
+                    VkFormatFeatureFlagBits2 is basically compensating for the fuckup when `VkFormatFeatureFlagBits` could only have 31 flags
+                    this type is VkFlags64 and added two extra flags, namely:
+                        1. VK_FORMAT_FEATURE_2_STORAGE_READ_WITHOUT_FORMAT_BIT_KHR and VK_FORMAT_FEATURE_2_STORAGE_WRITE_WITHOUT_FORMAT_BIT_KHR indicate that an implementation supports respectively reading and writing a given VkFormat through storage operations without specifying the format in the shader.
+                        2. VK_FORMAT_FEATURE_2_SAMPLED_IMAGE_DEPTH_COMPARISON_BIT_KHR indicates that an implementation supports depth comparison performed by OpImage*Dref* instructions on a given VkFormat. Previously the result of executing a OpImage*Dref* instruction on an image view, where the format was not one of the depth/stencil formats with a depth component, was undefined. This bit clarifies on which formats such instructions can be used.
```

Comment:
mind formatting the comment so I can read it on a 1080p screen without sie scrolling?

ok we should always use `VkFormatProperties3` then if its core in Vk 1.1 (our minimum required)

Reply:
Yes